File: c/phonebook.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

typedef struct // data struct creation
{
    string name; // key
    string number; // key
}
person; // struct name

int main(void)
{
    person people[3];

    people[0].name = "Carter";
    people[0].number = "+1-617-495-1000";

    people[1].name = "David";
    people[1].number = "+1-617-495-1000";

    people[2].name = "John";
    people[2].number = "+1-617-468-2750";

    string name = get_string("Name: "); // search for name

    for (int i = 0; i < 3; i++)
    {
        if (strcmp(people[i].name, name) == 0) // getting name
        {
            printf("Found %s\n", people[i].number); // getting number
            return 0;
        }
    }
    printf("Not found\n");
    return 1;
}

File: python/phonebook.py
# here comes a list, inside of it are dictionaries

""" people = [
    {"name": "Carter", "number": "+1-617-495-1000"},
    {"name": "David", "number": "+1-617-495-1000"},
    {"name": "John", "number": "+1-949-468-2750"},
] """

people = {
    "Carter" : "+1-617-495-1000",
    "David" : "+1-617-495-1000",
    "John" : "+1-949-468-2750"
}

# people is now a list of dict (dictionaries)

name = input("Name: ")

if name in people:
    number = people[name]
    print(f"Found {number}")
else:
    print("Not found")
